src/main/kotlin/uk/gov/dluhc/printapi/database/entity/Certificate.kt

```diff
@@ -66,6 +66,11 @@ class Certificate(
     @field:NotNull
     var suggestedExpiryDate: LocalDate = issueDate.plusYears(10),
 
+    /**
+     * Certificate status corresponds to the current status of the most recent
```

Comment:
I assume from the implementation, this will be the status value of the last PrintRequestStatus that is added to certificate's print requests. If that is the case and we assume we can receive a new print request without the previous one being completely processed, we might end up with a status of a prior request when it is updated.

Reply:
```
    private fun assignStatus() {
        val currentPrintRequest = getCurrentPrintRequest()
        val currentStatus = currentPrintRequest.getCurrentStatus()
        status = currentStatus.status
    }
```
The status relates to the current print request that selects the PrintRequest with the most recent `requestDateTime` and then for that print request the latest status.

So I don't the scenario you described will occur.  Any "previous" print requests that are updated do not affect the certificate status.

Reply:
Thanks, @Neil-Massey ,
I saw the assignStatus implementation, then going back and forth with the other code, I forgot how it worked. You're right. You can ignore this comment.